web/libs/editor/src/stores/AppStore.js

```diff
@@ -859,10 +859,25 @@ export default types
       self.users.replace(users);
     }
 
+    // @deprecated use `enrichUsers` instead (as mergeUsers will not update existing users and can lose previous data)
     function mergeUsers(users) {
       self.setUsers(uniqBy([...getSnapshot(self.users), ...users], 'id'));
     }
 
+    function enrichUsers(users) {
+      const oldUsers = getSnapshot(self.users);
+      const oldUsersMap = {};
+      oldUsers.forEach(user => {
+        oldUsersMap[user.id] = user;
+      });
+      const newUsers = users.map(user => {
+        const oldUser = oldUsersMap[user.id];
+
+        return oldUser ? { ...oldUser, ...user } : user;
```

Comment:
This can be simplified to not create too much extra checks and vars
```suggestion
        return { ...oldUsersMap[user.id], ...user };
```